x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -2937,6 +2957,56 @@ public void testPruneRedundantSortClausesUsingAlias() {
         );
     }
 
+    public void testInsist_fieldExists_noUnmappedFieldsAnywhere() {
```

Comment:
This test is only looking at what the Analyzer does. It shouldn't be in this class.